packages/teams-js/src/private/files.ts

```diff
@@ -529,4 +593,206 @@ export namespace files {
 
     sendMessageToParent('files.openDownloadFolder', [fileObjectId], callback);
   }
+
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Initiates add 3P cloud storage provider flow
+   *
+   * @param callback Callback that will be triggered post add 3P cloud storage provider action
+   */
+  export function add3PCloudStorageProvider(callback: (error?: SdkError) => void): void {
+    ensureInitialized(FrameContexts.content);
+
+    if (!callback) {
+      throw new Error('[files.add3PCloudStorageProvider] callback cannot be null');
+    }
+
+    sendMessageToParent('files.add3PCloudStorageProvider', [], callback);
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Initiates signout of 3P cloud storage provider flow
+   *
+   * @param callback Callback that will be triggered post signout of 3P cloud storage provider action
+   */
+  export function remove3PCloudStorageProvider(
+    logoutRequest: I3PCloudStorageProviderRequest<I3PCloudStorageProviderLogoutRequestContentType>,
+    callback: (error?: SdkError) => void,
+  ): void {
+    ensureInitialized(FrameContexts.content);
+
+    if (!validateNonEmptyContent(logoutRequest, callback)) {
+      return;
+    }
+    if (!validateArgumentConsistency(logoutRequest, callback)) {
```

Comment:
Shouldn't we be throwing an SdkError rather than silently returning if no callback is specified?

Reply:
Either that or shouldn't we be throwing if there is no callback ahead of these two checks? Otherwise if you don't speciy a callback I think this will just silently do nothing

Reply:
This piece of code has been removed since it no longer is required, hence the comment is no longer applicable.